include/linux/tee_drv.h

```diff
@@ -586,6 +593,48 @@ static inline bool tee_param_is_memref(struct tee_param *param)
 	}
 }
 
+static inline bool tee_ioctl_param_is_ocall_reply(struct tee_ioctl_param *param)
+{
+	u64 type = param->attr & TEE_IOCTL_PARAM_ATTR_TYPE_MASK;
+
+	return param->attr & TEE_IOCTL_PARAM_ATTR_OCALL &&
+	       type == TEE_IOCTL_PARAM_ATTR_TYPE_VALUE_INOUT &&
+	       param->a;
+}
+
+static inline bool tee_param_is_ocall_request(struct tee_param *param)
+{
+	u64 type = param->attr & TEE_IOCTL_PARAM_ATTR_TYPE_MASK;
+
+	return param->attr & TEE_IOCTL_PARAM_ATTR_OCALL &&
+	       type == TEE_IOCTL_PARAM_ATTR_TYPE_VALUE_INOUT &&
+	       param->u.value.a;
+}
+
+static inline bool tee_param_is_ocall_request_safe(struct tee_param *param)
+{
+	return param ? tee_param_is_ocall_request(param) : false;
+}
+
+static inline bool tee_param_is_ocall(struct tee_param *param)
+{
+	u64 type = param->attr & TEE_IOCTL_PARAM_ATTR_TYPE_MASK;
+
+	return param->attr & TEE_IOCTL_PARAM_ATTR_OCALL &&
+	       type == TEE_IOCTL_PARAM_ATTR_TYPE_VALUE_INOUT;
+}
+
+static inline void tee_param_clear_ocall(struct tee_param *ocall)
```

Comment:
This can be moved into `drivers/tee/optee/call.c`

Reply:
Moved.